stix_shifter_modules/sentinelone/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,411 @@
+    """
+
+    def __init__(self, pattern: Pattern, data_model_mapper, options):
+        self.dmm = data_model_mapper
+        self.comparator_lookup = self.dmm.map_comparator()
+        self.pattern = pattern
+        self.options = options
+        self.timeframe = []
+        self.type_map = self.load_json(TYPE_MAP_PATH)
+        self.translated = self.parse_expression(pattern)
+
+    @staticmethod
+    def load_json(rel_path_of_file):
+        """
+        Consumes a json file and returns a dictionary
+        :param rel_path_of_file: str
+        :return: dict
+        """
+        _json_path = path.dirname(path.realpath(__file__)) + "/" + rel_path_of_file
+        if path.exists(_json_path):
+            with open(_json_path, encoding='utf-8') as f_obj:
+                return json.load(f_obj)
+        raise FileNotFoundError
+
+    @staticmethod
+    def _check_boolean_value(final_value):
+        """
+        returns boolean value of input
+        :param final_value:str
+        :return bool
+        """
+        if final_value.lower() == "true" or (final_value.isdigit() and final_value == "1"):
+            boolean_value = bool(True)
+        elif final_value.lower() == "false" or (final_value.isdigit() and final_value == "0"):
+            boolean_value = bool(False)
+        else:
+            raise NotImplementedError('Invalid boolean type input')
+        return boolean_value
+
+    def _check_field_value_support(self, filter_name, value):
+        """
+        check input value format that matches with the field supported values
+        """
+        final_value = str(value)
+        if filter_name in self.type_map["enum_supported_fields"]:
+            if final_value.upper() not in [x.upper() for x in self.type_map["enum_supported_values"][filter_name]]:
+                raise NotImplementedError(f'Unsupported ENUM values provided. Possible supported enum values are'
+                                          f'{self.type_map["enum_supported_values"][filter_name]}')
+            final_value = final_value.upper()
+        if filter_name in self.type_map["int_supported_fields"]:
+            if not final_value.isdigit():
+                raise NotImplementedError(f'string type input - {value} is not supported for int field - {filter_name}')
+            final_value = str(value)
+        elif filter_name in self.type_map["boolean_supported_fields"]:
+            final_value = QueryStringPatternTranslator._check_boolean_value(final_value)
+        return final_value
+
+    def _check_value_comparator_support(self, exp_comparator, filter_name):
+        """
+        check operator format that matches with the field supported values
+        """
+        filter_type = str(exp_comparator).split(".")[1]
+        if filter_name in ["fileFullName"] and filter_type in \
+                ["LessThan", "LessThanOrEqual", "GreaterThan", "GreaterThanOrEqual", "NotEqual", "Equal"]:
```

Comment:
Does this mean you can't do an equals search for a file name? It seems like an odd omission but I'm not sure how the data from `"fileFullName"` would look so it may make sense.

Reply:
We have converted equal operator into LIKE operator now for filename search.